src/hooks/useUsers.tsx

```diff
@@ -1,26 +1,58 @@
 import { UsersContext } from "UsersContext";
-import { useContext } from "react";
+import { useContext, useCallback } from "react";
 
 import { UserData } from "types";
+import api from "api";
 
 const useUsers = () => {
     const { users, updateUsers } = useContext(UsersContext);
 
-    const addUser = (values: UserData) => {
-        updateUsers([...users, values]);
-    };
-
-    const retrieveUser = (id: number) => users.find((v) => v.id === id);
-
-    const updateUser = (id: number, user: Partial<UserData>) => {
-        updateUsers(users.map((v) => (v.id === id ? { ...v, ...user } : v)));
-    };
-
-    const deleteUser = (id: number) => {
-        updateUsers(users.filter((v) => v.id !== id));
-    };
-
-    return { users, addUser, retrieveUser, updateUser, deleteUser };
+    const getUser = (id: number) => users.find((v) => v.id === id);
+
+    const retrieveUsers = useCallback(() => {
+        updateUsers([]);
+        return api.get("users").then((data: UserData[]) => {
+            updateUsers(data);
+        });
+      }, [updateUsers]);
+    
+      const addUser = useCallback(
+        (values: Omit<UserData, "id">) => {
+          return api.post("users", values).then((data: UserData) => {
+            updateUsers([...users, data]);
```

Comment:
let's use "callback" version of `updateUsers` to get rid of `users` dependency
```suggestion
            updateUsers(prevUsers => [...prevUsers, data]);
```